demo/config/targets/INST/screens/commanding.txt

```diff
@@ -36,7 +36,7 @@ VERTICAL
     # Here as an example we're using the BACKGROUNDBUTTON which runs all the code in the background.
     # Note we didn't need to use this here but it's an example of wrapping calls to the GUI in Qt.execute_in_main_thread.
     # You need to use this button if your code blocks or takes a long time to execute so you don't hang the screen.
-    BACKGROUNDBUTTON 'Run Script' 'Cosmos.run_process("#{Cosmos::USERPATH}/tools/ScriptRunner -r #{Qt.execute_in_main_thread { get_named_widget("SCRIPTNAME").text }}")'
+    BACKGROUNDBUTTON 'Run Script' 'script = ""; Qt.execute_in_main_thread { script = get_named_widget("SCRIPTNAME").text }; Cosmos.run_process("#{Cosmos::USERPATH}/tools/ScriptRunner -r #{script}")'
```

Comment:
This ends up running ScriptRunner.bat on Windows which causes unexpected text output from the launcher.  Probably needs to be ruby ScriptRunner or something fancier.

Reply:
It's calling Cosmos.run_process with ScriptRunner. That shouldn't use a batch file.

Reply:
run_process just runs a process.  Ruby isn't specified so the command shell is going to run the bat file.